src/fairmat_readers_xrd/readers.py

```diff
@@ -40,6 +40,17 @@ def transfer_data_into_template(**kwargs):
     raise NotImplementedError
 
 
+def read_file(file_path: str) -> dict:
+    """Main function to figure out which specific file format to read."""
+    if file_path.endswith(".rasx"):
+        return read_rigaku_rasx(file_path)
+    if file_path.endswith(".xrdml"):
+        return read_panalytical_xrdml(file_path)
+    if file_path.endswith(".brml"):
+        return read_bruker_brml(file_path)
+    return {}

```

Comment:
Let's raise an error (or at least a warning along with the return) in case none of the readers is compatible with the file extension.
```suggestion
    raise NotImplementedError('Unknown file type.')
```